rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake

```diff
@@ -142,6 +143,9 @@ target_link_libraries(${rosidl_generate_interfaces_TARGET}${_target_suffix} PUBL
   rosidl_runtime_c::rosidl_runtime_c
   rosidl_typesupport_interface::rosidl_typesupport_interface
   rcutils::rcutils)
+add_dependencies(
```

Comment:
Following a pattern can be good but also there is a push for modern CMake targets across the ROS2 code, so I'd personally be fine with going that way. What would that look like here? I'm a bit behind the times on CMake

Reply:
If I'm not wrong, you'd add `${rosidl_generate_interfaces_TARGET}__rosidl_generator_type_hash::${rosidl_generate_interfaces_TARGET}__rosidl_generator_type_hash` as a target to target_link_libraries.

(It looks pretty ugly though, because it's so long :shrug:)

Reply:
I'm not 100% sure but I don't think I can use `add_custom_command` target in `target_link_libraries` - I think even the modern way to achieve this particular result is to use `add_dependencies` because the target is not a library.

Reply:
Ahh, I didn't realize this was a custom target. No issues then!